h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -279,6 +293,38 @@ public static long parseTimeNanos(String s, int start, int end) {
         return ((((hour * 60L) + minute) * 60) + second) * NANOS_PER_SECOND + nanos;
     }
 
+    /**
+     * Returns the current value of the high-resolution time source plus the
+     * specified offset.
+     *
+     * @param ms
+     *            additional offset in milliseconds
+     * @return time in nanoseconds, never equal to 0
+     * @see System#nanoTime()
+     */
+    public static long currentNanoTimePlusMillis(int ms) {
+        return nanoTimePlusMillis(System.nanoTime(), ms);
+    }
+
+    /**
+     * Returns the current value of the high-resolution time source plus the
+     * specified offset.
+     *
+     * @param nanoTime
+     *            time in nanoseconds
+     * @param ms
+     *            additional offset in milliseconds
+     * @return time in nanoseconds, never equal to 0
+     * @see System#nanoTime()
+     */
+    public static long nanoTimePlusMillis(long nanoTime, int ms) {
```

Comment:
`DateTimeUtils` is for various datetime operations in SQL engine and not for common methods moved here, old their location was correct.